crates/node/src/protocol.rs

```diff
@@ -584,149 +611,169 @@ where
+                    "Transfer of tx fee cannot be applied because of an \
+                     error: {}. This shouldn't happen.",
+                    e
+                );
+                return Err(e.into());
+            }
+        }
+    };
 
-            Ok(valid_batched_tx_result)
+    const FEE_PAYMENT_DESCRIPTOR: std::borrow::Cow<'static, str> =
+        std::borrow::Cow::Borrowed("wrapper-fee-payment");
+    let current_block_height = shell_params
+        .state
+        .in_mem()
+        .get_last_block_height()
+        .next_height();
+
+    // Event for the tip
+    let proposer_post_balance = Some(
+        token::read_balance(
+            shell_params.state,
+            &wrapper.fee.token,
+            block_proposer,
+        )
+        .map_err(Error::Error)?
+        .into(),
+    );
+    shell_params.state.write_log_mut().emit_event(
+        TokenEvent {
+            descriptor: FEE_PAYMENT_DESCRIPTOR,
+            level: EventLevel::Tx,
+            operation: TokenOperation::transfer(
+                UserAccount::Internal(wrapper.fee_payer()),
+                UserAccount::Internal(block_proposer.clone()),
+                wrapper.fee.token.clone(),
+                fee_components.tip.into(),
+                post_bal.into(),
+                proposer_post_balance,
+            ),
         }
-        Err(e) => {
-            // Fee overflow. This shouldn't happen as it should be prevented
-            // by process_proposal.
-            tracing::error!(
-                "Transfer of tx fee cannot be applied to due to fee overflow. \
-                 This shouldn't happen."
-            );
-            Err(Error::FeeError(format!("{}", e)))
+        .with(HeightAttr(current_block_height))
+        .with(TxHashAttr(tx.header_hash())),
+    );
+    // Event for the base fee
+    let operation = if wrapper.fee.token
+        == shell_params
+            .state
+            .get_native_token()
+            .map_err(Error::Error)?
+    {
+        TokenOperation::Burn {
+            target_account: UserAccount::Internal(wrapper.fee_payer()),
+            token: wrapper.fee.token.clone(),
+            amount: fee_components.base.into(),
+            post_balance: post_bal.into(),
         }
-    }
+    } else {
+        let pgf_post_balance = Some(
+            token::read_balance(
+                shell_params.state,
+                &wrapper.fee.token,
+                &namada_sdk::address::PGF,
+            )
+            .map_err(Error::Error)?
+            .into(),
+        );
+        TokenOperation::transfer(
+            UserAccount::Internal(wrapper.fee_payer()),
+            UserAccount::Internal(namada_sdk::address::PGF),
+            wrapper.fee.token.clone(),
+            fee_components.base.into(),
+            post_bal.into(),
+            pgf_post_balance,
+        )
+    };
+    shell_params.state.write_log_mut().emit_event(
+        TokenEvent {
+            descriptor: FEE_PAYMENT_DESCRIPTOR,
```

Comment:
```suggestion
            descriptor: FEE_PAYMENT_DESCRIPTOR,
```
should we use a different descriptor here? `wrapper-fee-treasury` or smth, idk

Reply:
We can do that yes, should we also create a new one for when we burn the base fee (native token)? Maybe we can just create two descriptors, one for the tip and one for the base fee. It looks like no external tool is using these events so even changing them shouldn't require modifications anywhere else

Reply:
yeah, I think one for tips and another for base fees is enough

Reply:
I've split the event descriptor in 93de5ad128. I've also changed the logic to emit the tip event only if a tip is present